.github/actions/install-nix-action/action.yaml

```diff
@@ -37,14 +43,57 @@ runs:
 
         gh run download "$RUN_ID" --repo "$DOGFOOD_REPO" -n "$INSTALLER_ARTIFACT" -D "$INSTALLER_DOWNLOAD_DIR"
         echo "installer-path=file://$INSTALLER_DOWNLOAD_DIR" >> "$GITHUB_OUTPUT"
+        TARBALL_PATH="$(find "$INSTALLER_DOWNLOAD_DIR" -name 'nix*.tar.xz' -print | head -n 1)"
+        echo "tarball-path=file://$TARBALL_PATH" >> "$GITHUB_OUTPUT"
 
         echo "::notice ::Dogfooding Nix installer from master (https://github.com/$DOGFOOD_REPO/actions/runs/$RUN_ID)"
       env:
         GH_TOKEN: ${{ inputs.github_token }}
         DOGFOOD_REPO: "NixOS/nix"
+    - name: "Download experimental installer"
+      shell: bash
+      id: download-experimental-nix-installer
+      if: ${{ inputs.experimental-installer == 'true' }}
+      run: |
+        if [ "$RUNNER_OS" == "Linux" ]; then
+          INSTALLER_OS="linux"
+        elif [ "$RUNNER_OS" == "macOS" ]; then
+          INSTALLER_OS="darwin"
+        else
+          echo "::error ::Unsupported RUNNER_OS: $RUNNER_OS"
+        fi
+
+        if [ "$RUNNER_ARCH" == "X64" ]; then
+          INSTALLER_ARCH="x86_64"
+        elif [ "$RUNNER_ARCH" == "ARM64" ]; then
+          INSTALLER_ARCH="aarch64"
+        else
+          echo "::error ::Unsupported RUNNER_ARCH: $RUNNER_ARCH"
+        fi
+
+        EXPERIMENTAL_INSTALLER_ARTIFACT="nix-installer-$INSTALLER_ARCH-$INSTALLER_OS"
+        EXPERIMENTAL_INSTALLER_PATH="$GITHUB_WORKSPACE/$EXPERIMENTAL_INSTALLER_ARTIFACT"
+        # TODO: This uses the latest release. It should probably be pinned, or dogfood the experimental repo's default branch - similar to the above
```

Comment:
One option would be to add another input and matrix item to allow you to target target both release and master. Though I'd probably add later as needed.